include/oneapi/tbb/detail/_concurrent_queue_base.h

```diff
@@ -123,7 +123,7 @@ class micro_queue {
             page_allocator_traits::construct(page_allocator, p);
         }
 
-        if (tail_counter.load(std::memory_order_relaxed) != k) spin_wait_until_my_turn(tail_counter, k, base);
+        spin_wait_until_my_turn(tail_counter, k, base);
```

Comment:
Double check idiom often requires `acquire` on the first load. The code does not call `pause` if condition is met. See `spin_wait_until_my_turn` implementation.

Reply:
Then I am confused. I made the statement about call to `pause` because of the `spin_wait_until_my_turn`'s implementation, which is 212 lines below. We can see there that the first thing the function does is calling the parameterized constructor of `atomic_backoff`. Its implementation is:
```cpp
    // This constructor pauses immediately; do not use on hot paths!
    atomic_backoff(bool) : count(1) { pause(); }
```
So, it seems the code unconditionally pauses the CPU, which is not expected on hot paths.

Reply:
Used another constructor in spin_wait_until_my_turn